src/randgraphs.jl

```diff
@@ -148,6 +148,48 @@ function _try_creation(n::Int, k::Vector{Int}, rng::AbstractRNG)
     return edges
 end
 
+"""Creates a [Barabási–Albert model](https://en.wikipedia.org/wiki/Barab%C3%A1si%E2%80%93Albert_model) 
+random graph with `n` nodes is grown by attaching new nodes each with `m` edges that 
+are preferentially attached to existing nodes with high degree. Undirected graphs are
+created by default; use `is_directed=true` to override.
+"""
+function barabasi_albert(n::Integer, m::Integer; is_directed=false)
+	@assert(1<=m<n, "Barabási–Albert network must have 1 <= m < n")
+	g = is_directed ? DiGraph(n) : Graph(n)
+	# Target nodes for new edges
+	targets = collect(1:m)
```

Comment:
targets is an array to store the target nodes. It will be updated in line 183, thus we need the collect.